### PR TITLE
Woodcutter & Artificer Expansion

### PR DESCRIPTION
**Pull request by Nakalon**
## About The Pull Request

This PR serves to expand a bit the roles of the Woodcutter and Artificer by giving them some content!
The plan was not only make Woodcutters more important by giving wood another interesting use but also doing the same for the Artificer, which having just been ported is still in an odd place.

This is not yet all complete, these would probably represent tier 1 of 4 that I plan to work on.
But for now, this is what I have to show you!

### For the Woodcutter

Here we have some content for the Woodcutters, it's not much for now, but I consider it an improvement already!

#### Essence of Lumber

image

Essence of Lumber is a rare item that may spawn based on woodcutting skill and Luck (So I advise you have a Seelie companion!)
This Essence of Lumber will be necessary for Artificer recipes, some that are important to the Woodcutter in specific such as further cart upgrades.

(A mob that would have been summoned using this new essence had to be cut down for now, but is still planed to make his way into the game once the theme is properly set in stone!)

#### Wood Planks

image

Holy fuck, amazement over Wooden Planks.
Now I know this might not seem like much but these planks would be important, and they create an important precedent:
Laborer - Refiner - Crafter
Here we're making the Woodcutter into the Laborer AND the Refiner, making him more important in the process and giving them just one more small thing to do, which may help with immersion and RP!

Yes for now these are made in the grinding wheel. Can't have everything in one go and I was already overextending myself sprite wise.

### For the Artificer

image

#### Artificer Table

image
image

Now Artificer is cool and all with its coppers and its fucking whatever else but now... Has any Artificer considered the power or raw... WOOD.
Yeah, wood's pretty sick, and now the Artificer has a table (which took me longer to code than it should have) just for his machinations.
The idea is Half of the Artificers production is Metals, the other half is wood! Wood is pretty cool yo!

Oh also they get a cool wooden mallet. (Which can be used for forging also because there were no other options for smithing hammers other than forging one... which is a bit hard if you have no hammer...)

image
image

#### Recipes

image

Together with this Table come some new recipes.
Just kidding, these are just recipes that were already in the game, they were relegated to the crafting tab under carpentry however, and you know how I hate that crafting tab.

These recipes, based on wood include arms and legs, some shields and wooden weapons such as the staff and the level 1 and level 2 upgrade cogs which can be used on the cart for increased capacity!

The level 2 cog requires essence of lumber! Treat your Woodcutters well!

image

#### Actually, I went fucking bonkers.

**ALL RECIPES THAT USED BRONZE OR ENGINEERING SKILL WERE MOVED FROM THE ANVIL TO THE ARTIFICER TABLE**

https://github.com/user-attachments/assets/21a145be-d887-425a-b847-f8a385feddba

#### Shunned Races

Now you can play the fucking role. Lets go!!!

### For both of them

So, do you know what useless square of marble in the middle of the town that served absolutely no purpose?
Now it does!
Its the Artificer and Woodcutter workshop!
Artificers will work and trade here while the woodcutters will saw their planks for them!

image

~~Much to my dismay I noticed I had to give them an anvil a wooden bin and a fucking forge, however I'd like to further separate them from smiths and do away with this in the future.~~
**HAUHAAUMAAUHAUHA I DID THIS ANYWAYS MUAHAHAHAH

image

image
image
image
image
image

As they moved, the Artificers took some things with them.
The locks were replaced however, smiths. Don't freak!
image

## Why It's Good For The Game

I like playing woodcutter.
However I can't play woodcutter without wanting to kill myself 1 hour in.
No one plays wood cutter because there's no content. I'm trying to add content.
The Artificer is also at an odd place having just arrived, this will hopefully give him something cool to fiddle with and they can become best friends.

## Changelog

map: Added Artificer workshop, removed some items from the former masonry and changed those locks to smithing
add: Added Essence of Lumber, a rare drop that may spawn based on lumberjack skill and luck when chopping wood
add: Added Wooden Planks, a result of processing small wooden logs that may be used by artificers
add: Added the Artificer Table and the Wooden Mallet
tweak: Moved all engineering recipes to the Artificer table.
tweak: Shifted some carpeting recipes over to Engineering and the new artificer table
balance: Reduced wood prices
code: Slightly improved Bomb code to make it easier on future coders adding more bomb subtypes

### CODE REVIEW

**Comment by Nakalon** [comment]
Still need to solve some issues with the recipes not working proper and to do the mapping part.

**Comment by Elli-Skala** [comment]
Wooden walking sticks being only craft able by one yeoman in town is abit strange to me

**Comment by Gyrandola** [comment]
you're precious

**Comment by Nakalon** [comment]
> Wooden walking sticks being only craft able by one yeoman in town is abit strange to me

Uh, actually I didn't delete that recipe! woo! Just duplicated it here!

**Comment by Gyrandola** [comment]
Consider adding thhe recipes I originally put in engineering.dm

**Comment by Nakalon** [comment]
> Consider adding the recipes I originally put in engineering.dm

We're so in sync, I'm doing that now!
Adding the necessary code to the Art table so it can accept the bloody metals!
No forge necessary!

**Comment by Nakalon** [comment]
This is ALMOST ALMOST ready!

Only have a recipe thats bugged and I don't get why...

**Comment by Nakalon** [comment]
This is ready but have to deal with Map Conflicts.

Also waiting on a review from Useroth!

**Comment by Nakalon** [comment]
Conflicts solved! Waiting on Useroths review!

**Comment by Gyrandola** [comment]
> Conflicts solved! Waiting on Useroths review!

Am I not good enough for you anymore

**Comment by Nakalon** [comment]
> > Conflicts solved! Waiting on Useroths review!
> 
> Am I not good enough for you anymore

Ofc you are and you may! But I asked Useroths help to debug and after helping me he asked if a very picky review would be helpful.

So I can better my code!

**Review by Gyrandola — APPROVED** [review]

**Comment by TheFenix323** [comment]
Is the artificer not supposed to be able to saw logs for planks?

**Comment by TheFenix323** [comment]
> > Is the artificer not supposed to be able to saw logs for planks?
> 
> The ability to saw the wood is based on lumberjack skill!
> 
> Artificers get 1 level, which gives them like 30% chance to fail that they can build on with skill if they have to.
> 
> Woodcutters get 4 so they get 0 failchance.

Then I had terrible luck testing the Artificer, used all the logs in the shop and got no planks. 😭

**Comment by Nakalon** [comment]
> > > Is the artificer not supposed to be able to saw logs for planks?
> > 
> > 
> > The ability to saw the wood is based on lumberjack skill!
> > Artificers get 1 level, which gives them like 30% chance to fail that they can build on with skill if they have to.
> > Woodcutters get 4 so they get 0 failchance.
> 
> Then I had terrible luck testing the Artificer, used all the logs in the shop and got no planks. 😭

This is very possible, I had a similar result and I thought I was just unlucky. Maybe the logic is fucked. Do you think you could check? I'll look for it too, 2 eyes are better than one, somethings fucky here...

**Comment by Nakalon** [comment]
> > > > Is the artificer not supposed to be able to saw logs for planks?
> > > 
> > > 
> > > The ability to saw the wood is based on lumberjack skill!
> > > Artificers get 1 level, which gives them like 30% chance to fail that they can build on with skill if they have to.
> > > Woodcutters get 4 so they get 0 failchance.
> > 
> > 
> > Then I had terrible luck testing the Artificer, used all the logs in the shop and got no planks. 😭
> 
> This is very possible, I had a similar result and I thought I was just unlucky. Maybe the logic is fucked. Do you think you could check? I'll look for it too, 2 eyes are better than one, somethings fucky here...

This was solved

**Review by Useroth — COMMENTED** [review]
Honestly looks fairly good otherwise.

File: code/game/objects/items/rogueitems/natural/wood.dm
Comment (Nakalon) [reply code/game/objects/items/rogueitems/natural/wood.dm:28]: Thanks! I'll do these manually to remember them!

**Pull request merged**
